sdk/cosmos/azure-cosmos/README.md

```diff
@@ -467,6 +467,45 @@ print(json.dumps(container_props['defaultTtl']))
 
 For more information on TTL, see [Time to Live for Azure Cosmos DB data][cosmos_ttl].
 
+### Using operation response headers
+
+We have recently made some quality of life enhancements to our item-level operations, allowing users to look at their operations' response headers directly from the response returned by the methods.
+Response headers include metadata information from the executed operations like `etag`, which allows for optimistic concurrency scenarios.
+These changes apply to all item operations in both the sync and async clients - and can be used by referencing the `response_headers` property of any response as such:
+```python
+from azure.cosmos import CosmosClient
+import os
+
+URL = os.environ['ACCOUNT_URI']
+KEY = os.environ['ACCOUNT_KEY']
+DATABASE_NAME = 'testDatabase'
+CONTAINER_NAME = 'products'
+client = CosmosClient(URL, credential=KEY)
+database = client.get_database_client(DATABASE_NAME)
+container = database.get_container_client(CONTAINER_NAME)
+
+operation_response = container.create_item({"id": "test_item", "productName": "test_item"})
+operation_headers = operation_response.response_headers
+```
+For queries, response headers will be present when using paging only:
+```python
+from azure.cosmos import CosmosClient
+import os
+
+URL = os.environ['ACCOUNT_URI']
+KEY = os.environ['ACCOUNT_KEY']
+DATABASE_NAME = 'testDatabase'
+CONTAINER_NAME = 'products'
+client = CosmosClient(URL, credential=KEY)
+database = client.get_database_client(DATABASE_NAME)
+container = database.get_container_client(CONTAINER_NAME)
+
+query_response = container.query_items("select * from c", enable_cross_partition_query=True)
+query_pages = query_response.by_page()
+first_page = query_pages.next()
```

Comment:
This by_page sample looks awkward to me - I would expect the flow to be more like:
```python
query_response = container.query_items(...)
for page in query_response.by_page():
    headers: Dict = page.get_response_headers()
    for item in page:
        print(item)
```
This format also is more intuitive for async.

Reply:
the only issue with the above is that the initial pageable will have no response headers - they only get populated as we start iterating through the pages. I was trying to make the distinction that they will only exist at that point, but I could make a mention of that here as well

Reply:
Why do the subsequent pages not have response headers? Each page represents an individual request right? So each one would be accompanied by headers?

Reply:
each page is just a list_iterator of the items in the page - the query iterable object that we use to extract the headers is the item_pages:
![image](https://github.com/user-attachments/assets/45c8295c-27b0-41e9-a105-1a037a6169f7)